src/auth/oidc/models/openid.class.spec.ts

```diff
@@ -62,3 +62,61 @@ test('OIDC configure initializeSession', () => {
     oidc.initializeSession()
     expect(spy).toBeCalled()
 })
+
+test('OIDC OptionsMapper', () => {
+    const options = {
+        authorizationURL: '',
+        tokenURL: '',
+        clientID: 'clientId',
+        clientSecret: 'Clientsecret',
+        discoveryEndpoint: 'someEndpoint',
+        issuerURL: 'issuer_url',
+        logoutURL: 'logouturl',
+        callbackURL: 'redirect_uri',
+        responseTypes: ['none'],
+        scope: 'some scope',
+        sessionKey: 'key',
+        tokenEndpointAuthMethod: 'client_secret_basic',
+        useRoutes: false,
+    }
+    const openIdOptions = oidc.getOpenIDOptions(options)
+
+    expect(openIdOptions.client_id).toEqual(options.clientID)
+    expect(openIdOptions.client_secret).toEqual(options.clientSecret)
+    expect(openIdOptions.discovery_endpoint).toEqual(options.discoveryEndpoint)
+    expect(openIdOptions.issuer_url).toEqual(options.issuerURL)
+    expect(openIdOptions.logout_url).toEqual(options.logoutURL)
+    expect(openIdOptions.redirect_uri).toEqual(options.callbackURL)
+    expect(openIdOptions.response_types).toEqual(options.responseTypes)
+    expect(openIdOptions.scope).toEqual(options.scope)
+    expect(openIdOptions.sessionKey).toEqual(options.sessionKey)
+    expect(openIdOptions.token_endpoint_auth_method).toEqual(options.tokenEndpointAuthMethod)
+    expect(openIdOptions.useRoutes).toEqual(options.useRoutes)
+})
+
+test('test validateOptions', () => {
+    const options = {
+        authorizationURL: '',
+        tokenURL: '454',
+        clientID: 'clientId',
+        clientSecret: 'Clientsecret',
+        discoveryEndpoint: 'someEndpoint',
+        issuerURL: 'issuer_url',
+        logoutURL: 'logouturl',
+        callbackURL: 'redirect_uri',
+        responseTypes: ['none'],
+        scope: 'some scope',
+        sessionKey: 'key',
+        tokenEndpointAuthMethod: 'client_secret_basic',
+        useRoutes: false,
+    }
+    try {
+        oidc.validateOptions(options)
+    } catch (error) {
+        expect(error.message).toContain('"authorizationURL" is not allowed to be empty')
+    }
+
+    //positive case
+    options.authorizationURL = 'something'
+    oidc.validateOptions(options)
```

Comment:
can we 'actually' test this? like perhaps expect(validateOptions).not.toThrow or something?

Reply:
@paddybasi  done